types/dec_coin.go

```diff
@@ -157,9 +157,20 @@ func (coin DecCoin) IsValid() bool {
 // DecCoins defines a slice of coins with decimal values
 type DecCoins []DecCoin
 
-// NewDecCoins constructs a new coin set with decimal values
+// NewDecCoins constructs a new coin set with with decimal values
+// from DecCoins.
+func NewDecCoins(decCoinArray ...DecCoin) DecCoins {
+	decCoins := make(DecCoins, len(decCoinArray))
+	for i, decCoin := range decCoinArray {
+		decCoins[i] = decCoin
+	}
+
+	return decCoins
+}
+
+// NewDecCoinsFromCoin constructs a new coin set with decimal values
 // from regular Coins.
-func NewDecCoins(coins Coins) DecCoins {
+func NewDecCoinsFromCoin(coins Coins) DecCoins {
```

Comment:
since we're introducing this change, we can update it to
```suggestion
func NewDecCoinsFromCoin(coins ...Coin) DecCoins {
```